src/app/core/services/localeService.ts

```diff
@@ -1,24 +1,41 @@
 import { Injectable } from '@angular/core';
-import { appConfig } from 'src/app/shared/helpers/config';
+import { appConfig, Language } from 'src/app/shared/helpers/config';
+
 
 @Injectable({
   providedIn: 'root'
 })
 export class LocaleService {
 
-  readonly languages?: { tag: string, path: string }[];
-  readonly currentTag?: string;
+  readonly languages: Language[];
+  readonly currentTag: string;
+  readonly currentTagError: boolean
 
   constructor() {
     this.languages = appConfig.languages;
-    this.currentTag = this.languages?.find(l => window.location.pathname.endsWith(l.path))?.tag;
+    const fallbackTag = appConfig.production ? '' : 'en';
+    this.currentTag = this.languages.find(l => window.location.pathname.endsWith(l.path))?.tag ?? fallbackTag;
+    this.currentTagError = !this.currentTag;
   }
 
   navigateTo(langTag: string): void {
-    const nextLang = this.languages?.find(l => l.tag === langTag);
-    const currentLang = this.languages?.find(l => l.tag === this.currentTag);
+    const nextLang = this.languages.find(l => l.tag === langTag);
+    const currentLang = this.languages.find(l => l.tag === this.currentTag);
+
     if (!nextLang || !currentLang) throw new Error('Cannot find new or current lang in configured languages');
-    window.location.href = `${window.location.pathname.replace(currentLang.path, nextLang.path)}${window.location.hash}`;
+
+    const pathname = appConfig.production
+      ? this.getNextPathname(currentLang, nextLang)
+      : this.getNextPathnameForDevelopment(nextLang);
+    window.location.href = `${pathname}${window.location.hash}`;
+  }
+
+  private getNextPathnameForDevelopment(nextLang: Language): string {
+    return nextLang.path;
+  }
+
+  private getNextPathname(currentLang: Language, nextLang: Language): string {
+    return window.location.pathname.replace(currentLang.path, nextLang.path);
```

Comment:
This change might not be in the scope of the PR, I'll detail it shortly.

When changing the language, the app is reloaded by replacing in the URL the current tag by next tag. But in development, the current tag is not in the URL, as a consequence the URL computed for next tag is exactly the same, causing the app not to be reloaded.

I will need this change for the last step (which is displaying a language mismatch dialog), since this PR is quite small I assumed it was ok to do it here.